### PR TITLE
Add cancer recall to set for AB#16084

### PR DESCRIPTION
**Pull request by BronzBierd**
# Implements [AB#16084](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16084)

## Description
1. Remove recall filtering on returned data from PHSA patient.
2. Removed obsolete test related to filtering out recall results.
3. Added a UI test to test for the differences between screening types.
4. Update data set description for QuickLink and Landing page

## Testing

- [X] Unit Tests Updated
- [X] Functional Tests Updated
- [ ] Not Required

image

image

## UI Changes
image

image

## Notes
Data now extended to include recall data, with eventType to diferentiate:

image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            ],
+        });
+        cy.login(
+            Cypress.env("keycloak.username"),
+            Cypress.env("keycloak.password"),
+            AuthMethod.KeyCloak
+        );
+        cy.checkTimelineHasLoaded();
+    });
+    it("Should display different cards for different types", () => {
```
Comment (BrianMaki) [inline Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js:37]: I see there is a UI test to show the two event types but is there an e2e test to validate the download of the files for the two event types?

Comment (BronzBierd) [reply Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js:37]: This may be brittle though but I'll add it.

**Comment by BronzBierd** [comment]
Updated test run: 
image

File: Apps/Patient/src/Models/BcCancerScreeningType.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /// <summary>
+    /// BC Cancer screening types.
+    /// </summary>
+    [JsonStringEnumMemberConverterOptions(deserializationFailureFallbackValue: Unknown)]
+    [JsonConverter(typeof(JsonStringEnumMemberConverter))]
+    public enum BcCancerScreeningType
+    {
+        /// <summary>
+        /// Unknown result from PHSA
```
Comment (MikeLyttle) [inline Apps/Patient/src/Models/BcCancerScreeningType.cs:28]:
```suggestion
        /// Unknown result from PHSA.
```

File: Testing/functional/tests/cypress/integration/e2e/timeline/bcCancerScreening.js [hunk 3]
```diff
@@ -31,46 +66,17 @@ describe("Diagnostic Imaging", () => {
...
+    });
 
-                        cy.verifyDownload(`bc_cancer_screening_${dateSuffix}`, {
-                            contains: true,
-                        });
-                    });
-            });
+    it("Validate recall file download", () => {
+        cy.get("[data-testid=timelineCard]")
+            .filter(`:contains(BC Cancer Screening)`)
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/e2e/timeline/bcCancerScreening.js:78]:
```suggestion
            .filter(`:contains("BC Cancer Screening")`)
```

It feels off without quotes in there.  I assume it functions the same.

File: Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js (hunk 1 above)
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js:37]:
```suggestion
    });
    
    it("Should display different cards for different types", () => {
```

File: Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const { AuthMethod } = require("../../../support/constants");
+
+describe("BC Cancer Screening cards", () => {
+    function testCard(cardTitle, cardButtonText) {
+        cy.get("[data-testid=timelineCard")
+            .filter(`:contains(${cardTitle})`)
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/timeline/bcCancerScreening.js:6]:
```suggestion
            .filter(`:contains("${cardTitle}")`)
```

File: Apps/WebClient/src/ClientApp/src/models/timeline/bcCancerScreeningTimelineEntry.ts [hunk 5]
```diff
@@ -1,35 +1,63 @@
...
             EntryType.BcCancerScreening,
-            new DateWrapper(model.resultDateTime, { isUtc: true })
+            new DateWrapper(date, { isUtc: true })
         );
-
-        this.title = "BC Cancer Result";
-        this.documentType = "Screening results";
-        this.programName = model.programName;
+        this.isResult = isResult;
+        this.entryDate = date;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/models/timeline/bcCancerScreeningTimelineEntry.ts:35]: I think the `entryDate` property can be removed.  If anything is using it, they could swap to using the `date` property (from the parent class) instead.

File: Apps/WebClient/src/ClientApp/src/models/timeline/bcCancerScreeningTimelineEntry.ts [hunk 6]
```diff
@@ -1,35 +1,63 @@
...
+        this.subtitle = `Programe: ${model.programName}`;
         this.fileId = model.fileId;
-        this.resultDate = model.resultDateTime;
+        this.screeningType = model.eventType;
+        this.fileName = this.isResult
+            ? "bc_cancer_result"
+            : "bc_cancer_screening";
+        this.eventText = this.isResult
+            ? "BC Cancer Result PDF"
+            : "BC Cancer Screening PDF";
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/models/timeline/bcCancerScreeningTimelineEntry.ts:44]: These assignments could be moved into the `setEntryProperties()` function so they're grouped with the other properties that change based on event type.

**Review by BrianMaki — APPROVED** [review]

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**